### DISCUSSION TITLE
Requirements

### DISCUSSION THREAD

**Discussion by OzzyTheGiant** [medusajs/medusa#1001] [Q&A]
Just discovered this repo and I have to say this work looks amazing! Kudos to all involved. That being said, I have questions about the requirements for this platform. The docs mention that Redis and PostgreSQL are needed. These tools are excellent choices but there is one thing that bugs me in particular. It seems that these choices lock developers down to using either dedicated or cloud servers; I'm sure that was the intended goal to begin with, but given that this is an open source platform it would be ideal if it had wider support for different hosting environments.

To give a bit more context, dedicated, cloud and even most VPS servers allow full control of installation of these tools. However, some VPS environments are fully managed by the hosting company and limit what you can install on them (particularly Dreamhost). On such VPS environments, Node.js apps can be run but the choice of database is limited to MySQL, unless you opt to upgrade to a cloud server where you can install PostgreSQL. Given that the VPS is managed, sudo commands are not allowed and thus, Redis is not available. One can make the argument that we can simply host these databases elsewhere and just keep the VPS but that would incur more costs and there's additional security implications to consider. If the client is not willing to upgrade/move hosting accounts, the platform can't be used.

My question is, although not supported at this time, would it be possible that in the future, MySQL or even SQLite be supported in the future and would it be possible to make Redis optional, assuming the features that require Redis are not critical? I know it sounds like a lot to ask for, but I ask this because there are many small mom-and-pop shops out there that would like to keep hosting cheap and/or do not need the scale of a large business; given this platform is scalable, it would be ideal if the platform can scale sideways by supporting multiple databases, and by expanding coverage to the aforementioned restricted environments, it would be possible for me to use this platform and promote it more, convincing others to migrate away from WooCommerce, which as far as I can tell, is the only popular E-Commerce open source platform that can be hosted anywhere.

If not, what would it take to reach that level, if possible at all? I've been evaluating other platforms but they all seem to require the same thing (especially Saleor which has had my eye for a while but has heavy requirements that are not installable on a restricted VPS).

**Comment by kawpii** [comment]
By default it uses SQLite : https://docs.medusajs.com/troubleshooting/transaction-error-in-checkout/
